### DISCUSSION TITLE
the Certificate's default expiration time of K3S is one year，is there any parameters to adjust it.

### DISCUSSION THREAD

**Discussion by zhaofengcumt** [k3s-io/k3s#4907] [Q&A]
the Certificate's default expiration time of K3S is one year，is there any parameters to adjust it.

**Comment by briandowns** [comment]
Hi @zhaofengcumt ,

There's currently no way to adjust the certificates expiration date explicitly. That's a feature we're working on now actually. I don't know what release to expect it in though.

K3s will detect when certificates are nearly expiration and will update them for you. If you don't want to wait or need to perform a rottion manually. you can use the newly added subcommand (`certificate`) and force a certificate rotation that will regenerate the certificates from that point forward.

The command below will rotate all certificates.

```sh
k3s certificate rotate 
```

The command below will rotate certificates for a given service.

```sh
k3s certificate rotate --service api-server
```

**Reply by flaviogrossi** [reply]
Hi, i noticed that certificate expiration date is tunable by using the `CATTLE_NEW_SIGNED_CERT_EXPIRATION_DAYS` environment variable from the rancher/dynamiclistener project.
However not all certificates respect that, e.g. [here](https://github.com/k3s-io/k3s/blob/b0ed13485588651b1eba3cb0ba423c500325a28f/pkg/daemons/control/deps/deps.go#L592).

But it seems easy to add an ExpiresAt field to the cfg parameter to respect that environment variable (or to use another one).

Is this approach viable while #4651 is being implemented? Is that change enough? Will it be accepted?

Thanks

**Reply by flaviogrossi** [reply]
i opened a pull request here that should fix this problem: https://github.com/rancher/dynamiclistener/pull/61
